src/cantools/database/can/c_source.py

```diff
@@ -1205,148 +1223,153 @@ def _format_choices(signal, signal_name):
     return choices
 
 
-def _generate_encode_decode(message, use_float):
-    encode_decode = []
-
+def _generate_encode_decode(signal: "CodeGenSignal", use_float: bool) -> Tuple[str, str]:
     floating_point_type = _get_floating_point_type(use_float)
-    for signal in message.signals:
-        scale = float(signal.scale)
-        offset = float(signal.offset)
-
-        if offset == 0 and scale == 1:
-            encoding = 'value'
-            decoding = f'({floating_point_type})value'
-        elif offset != 0 and scale != 1:
-            encoding = f'(value - {offset}) / {scale}'
-            decoding = f'(({floating_point_type})value * {scale}) + {offset}'
-        elif offset != 0:
-            encoding = f'value - {offset}'
-            decoding = f'({floating_point_type})value + {offset}'
-        else:
-            encoding = f'value / {scale}'
-            decoding = f'({floating_point_type})value * {scale}'
 
-        encode_decode.append((encoding, decoding))
+    scale = signal.signal.scale
+    offset = signal.signal.offset
+
+    scale_literal = f"{scale}{'.0' if isinstance(scale, int) else ''}{'f' if use_float else ''}"
+    offset_literal = f"{offset}{'.0' if isinstance(offset, int) else ''}{'f' if use_float else ''}"
+
+    if offset == 0 and scale == 1:
+        encoding = 'value'
+        decoding = f'({floating_point_type})value'
+    elif offset != 0 and scale != 1:
+        encoding = f'(value - {offset_literal}) / {scale_literal}'
+        decoding = f'(({floating_point_type})value * {scale_literal}) + {offset_literal}'
+    elif offset != 0:
+        encoding = f'value - {offset_literal}'
+        decoding = f'({floating_point_type})value + {offset_literal}'
+    else:
+        encoding = f'value / {scale_literal}'
+        decoding = f'({floating_point_type})value * {scale_literal}'
 
-    return encode_decode
+    return encoding, decoding
 
 
-def _generate_is_in_range(message):
+def _generate_is_in_range(signal: "CodeGenSignal") -> str:
     """Generate range checks for all signals in given message.
 
     """
+    minimum = signal.signal.minimum
+    maximum = signal.signal.maximum
 
-    checks = []
+    if minimum is not None:
+        minimum = signal.signal.scaled_to_raw(minimum)
 
-    for signal in message.signals:
-        minimum = signal.minimum
-        maximum = signal.maximum
+    if maximum is not None:
+        maximum = signal.signal.scaled_to_raw(maximum)
 
-        if minimum is not None:
-            minimum = signal.scaled_to_raw(minimum)
+    if minimum is None and signal.minimum_value is not None:
```

Comment:
The names are IMO confusing here: what's the difference between `signal.minimum` and `signal.minimum_value` and `signal.minimum_type_value` without looking up the code of the respective properties? the latter two should probably be renamed to something like e.g. `minimum_raw_value` and `minimum_c_type_value`...

Reply:
I renamed the properties. I'm not sure, whether that logic here makes sense as it does not consider enums, but 🤷‍♂️